app/src/main/java/com/greenfox/tribesoflagopusandroid/fragments/MainFragment.java

```diff
@@ -6,21 +6,98 @@
+import com.greenfox.tribesoflagopusandroid.api.model.response.TroopsResponse;
+import com.greenfox.tribesoflagopusandroid.api.service.ApiService;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class MainFragment extends Fragment {
 
+    @Inject
+    ApiService apiService;
+
+    List<Building> buildings;
+    List<Resource> resources;
+    List<Troop> troops;
 
     public MainFragment() {
     }
 
-
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_main, container, false);
-    }
 
-}
+        TribesApplication.app().basicComponent().inject(this);
+
+        apiService.getResource(1).enqueue(new Callback<ResourcesResponse>() {
+            @Override
+            public void onResponse(Call<ResourcesResponse> call, Response<ResourcesResponse> response) {
+                resources = response.body().getResources();
+            }
+
+            @Override
+            public void onFailure(Call<ResourcesResponse> call, Throwable t) {
+
+            }
+        });
+
+        View rootView = inflater.inflate(R.layout.fragment_main, container, false);
+
+        ImageView goldImage = (ImageView) rootView.findViewById(R.id.gold_image);
+        ImageView foodImage = (ImageView) rootView.findViewById(R.id.food_image);
+        TextView gold = (TextView) rootView.findViewById(R.id.resources_gold);
+        TextView food = (TextView) rootView.findViewById(R.id.resources_food);
+        gold.setText(resources.get(0).getAmount() + " " + resources.get(0).getType());
+        food.setText(resources.get(1).getAmount() + " " + resources.get(1).getType());
+        goldImage.setImageResource(R.drawable.gold);
+        foodImage.setImageResource(R.drawable.food);
+
+        apiService.getBuildings(1).enqueue(new Callback<BuildingsResponse>() {
+            @Override
+            public void onResponse(Call<BuildingsResponse> call, Response<BuildingsResponse> response) {
+                buildings = response.body().getBuildings();
+            }
+
+            @Override
+            public void onFailure(Call<BuildingsResponse> call, Throwable t) {
+
+            }
+        });
+
+        TextView totalBuildingNumber = (TextView) rootView.findViewById(R.id.buildings_finished);
+        totalBuildingNumber.setText((buildings.size() + " finished"));
+
+        apiService.getTroops(1).enqueue(new Callback<TroopsResponse>() {
+            @Override
+            public void onResponse(Call<TroopsResponse> call, Response<TroopsResponse> response) {
+                troops = response.body().getTroops();
+            }
+
+            @Override
+            public void onFailure(Call<TroopsResponse> call, Throwable t) {
+
+            }
+        });
```

Comment:
@rekagiday these 3 info are in the `KingdomResponse` which would be one http request, now you have three